review_practice/react-app/src/containers/RoomList/ReservationModalBody.js

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useContext } from 'react';
+import { useMutation } from '@apollo/react-hooks';
+
+import * as Style from './style';
+
+import { SearchRoomContext } from 'pages/SearchRoom';
+
+import * as RoomAPI from 'graphql/mutation/booking';
+
+import { getNightValueFromDateString } from 'util/ConverDate';
+
+export default ({ room, searchOptions, closeModal }) => {
+    const { date, personnel } = searchOptions;
+    const { dispatchSearchOption } = useContext(SearchRoomContext);
+    const [addBooking, { data, loading, error }] = useMutation(RoomAPI.ADD_BOOKING);
+
+    useEffect(() => {
+        if (data && !data.reserveRoom.success) alert('이미 예약되어있는 숙소입니다');
+        if (data && data.reserveRoom.success) {
+            closeModal();
+            alert('예약이 완료됬습니다.');
+            dispatchSearchOption({ type: 'reset' });
+        }
+        if (loading) console.log('추가 중');
+        if (error) alert('이미 예약되어있는 숙소입니다');
+    }, [data, loading, error]);
+
+    const nights = date
+        ? getNightValueFromDateString({ startDateString: date.checkIn, endDateString: date.checkOut })
+        : 0;
+    const guest = personnel ? personnel.adult + personnel.children : 0;
+
+    const reserveButtonClick = () => {
+        const variables = { data: { roomId: room.id, date, personnel } };
+        addBooking({ variables });
+    };
+
+    return (
+        <Style.ReservationModalBody>
+            {loading && <small className="Loading-Statement">예약 처리중입니다. 잠시만 기달려주세요...</small>}
+            <h2>{room.name}</h2>
+            <hr></hr>
+            <label htmlFor="date">날짜</label>
+            {!date && <input className="Text-Center" id="date" type="text" value="날짜를 입력해주세요" readOnly />}
+            {date && (
+                <Style.DateWrapper id="date">
+                    <input className="Date-Input" id="start-date" type="text" value={date.checkIn} readOnly />
+                    <span className="Date-Arrow">></span>
+                    <input className="Date-Input" id="end-date" type="text" value={date.checkOut} readOnly />
+                </Style.DateWrapper>
+            )}
+            <label htmlFor="personnel">인원</label>
+            {personnel ? (
+                <input
+                    id="personnel"
+                    type="text"
+                    value={`게스트 ${guest}명${personnel.infant === 0 ? '' : `, 유아 ${personnel.infant}명`}`}
+                    readOnly
+                />
+            ) : (
+                <input className="Text-Center" id="personnel" type="text" value="인원을 설정해주세요" readOnly />
+            )}
+            <div className="Price-Wrapper">
+                {date && personnel ? (
```

Comment:
render 함수로 작성하는게 좋아 보입니다.
또한  data && personnel 이 아닌 변수를 만들어서 조금 더 명시적으로 표현하는게 좋아보입니다.

Reply:
```suggestion
                {isPersonnel ? (
```

isPersonnel 처럼 명시적인 변수를 만들어서 활용하면 가독성을 높일 수 있습니다.